src/imgbased/command.py

```diff
@@ -31,10 +30,12 @@ def nsenter(arg, new_root=None, shell=False, environ=None):
     environ = environ or os.environ
     log.debug("Executing: %s", arg)
     proc = subprocess.Popen(arg, stdout=subprocess.PIPE, env=environ,
-                            stderr=DEVNULL, shell=shell).communicate()
-    ret = proc[0]
-    log.debug("Result: %s", repr(ret))
-    return ret
+                            stderr=subprocess.PIPE, shell=shell)
+    stdout, stderr = proc.communicate()
+    log.debug("STDOUT: %s", repr(stdout))
+    log.debug("STDERR: %s", repr(stderr))
+    log.debug("ReturnCode: %s", repr(proc.returncode))
```

Comment:
Same, marked as resolved but I don't see any change in the code or reply.

Reply:
> This works but having single log like we have in vdsm can be more clear:
> 
> ```
> log.debug("Command %s terminated with return code %s, out %r, err %r",
>           arg, proc.returncode, out, err)
> ```
> 
> We don't need to use repr(obj), %r does it for us. There is no need to do repr of the return code, it is an integer.

Potentially the returncode can be None if for some weird reason the process hasn't been ended by the time we return from communicate(), so I preferred to use repr() on it as well.

Reply:
The return code cannot be None when communicate() returns. It returns after the process terminated and and the child was reaped.
https://docs.python.org/3.6/library/subprocess.html#subprocess.Popen.communicate

Regardless, logging None with %s is the same as logging with %r:

    >>> "%s %s %r %r" % (None, None, 0, 0)
   'None None 0 0'

You did not answer about using repr() instead of %r, I guess you missed this. This should be fixed.

Reply:
Since this is already merged, I'll open a new PR for that.

Reply:
Sent PR#17 ( https://github.com/oVirt/imgbased/pull/18 ) to address this.